File: SDK/Client/Python/src/raidtoolkit/AccountApi.py
from .RemoteApiClient import RemoteApiClient


class AccountApi:
    def __init__(self, client: RemoteApiClient):
        self.client = client

    def get_accounts(self):
        return self.client.call("account-api", "getAccounts")

    def get_account(self, accountId: str):
        return self.client.call("account-api", "accountInfo", [accountId])

    def get_account_dump(self, accountId: str):
        return self.client.call("account-api", "getAccountDump", [accountId])

    def get_all_resources(self, accountId: str):
        return self.client.call("account-api", "getAllResources", [accountId])

    def get_artifacts(self, accountId: str):
        return self.client.call("account-api", "getArtifacts", [accountId])

    def get_artifact_by_id(self, accountId: str, artifactId: str):
        return self.client.call("account-api", "getArtifactById", [accountId, artifactId])

    def get_heroes(self, accountId: str, snapshot: bool):
        return self.client.call("account-api", "getHeroes", [accountId, snapshot])

    def get_hero_by_id(self, accountId: str, heroId: str, snapshot: bool):
        return self.client.call("account-api", "getHeroById", [accountId, heroId, snapshot])

    def get_arena(self, accountId: str):
        return self.client.call("account-api", "getArena", [accountId])

    def get_academy(self, accountId: str):
        return self.client.call("account-api", "getAcademy", [accountId])

File: SDK/Client/DotNet/APIs/AccountApi.cs
using System;
using System.Reflection;
using System.Threading.Tasks;
using Raid.DataModel;
using Raid.Service;

namespace Raid.Client
{
    public class AccountApi : ApiBase<IAccountApi>, IAccountApi
    {
        internal AccountApi(RaidToolkitClient client) : base(client) { }

        public event EventHandler<SerializableEventArgs> Updated;

        public Task<Account> GetAccount(string accountId)
        {
            return CallMethod<Account>(MethodBase.GetCurrentMethod(), accountId);
        }

        public Task<RaidExtractor.Core.AccountDump> GetAccountDump(string accountId)
        {
            return CallMethod<RaidExtractor.Core.AccountDump>(MethodBase.GetCurrentMethod(), accountId);
        }

        public Task<Account[]> GetAccounts()
        {
            return CallMethod<Account[]>(MethodBase.GetCurrentMethod());
        }

        public Task<Resources> GetAllResources(string accountId)
        {
            return CallMethod<Resources>(MethodBase.GetCurrentMethod(), accountId);
        }

        public Task<Artifact> GetArtifactById(string accountId, int artifactId)
        {
            return CallMethod<Artifact>(MethodBase.GetCurrentMethod(), accountId, artifactId);
        }

        public Task<Artifact[]> GetArtifacts(string accountId)
        {
            return CallMethod<Artifact[]>(MethodBase.GetCurrentMethod(), accountId);
        }

        public Task<Hero> GetHeroById(string accountId, int heroId, bool snapshot = false)
        {
            return CallMethod<Hero>(MethodBase.GetCurrentMethod(), accountId, heroId, snapshot);
        }

        public Task<Hero[]> GetHeroes(string accountId, bool snapshot = false)
        {
            return CallMethod<Hero[]>(MethodBase.GetCurrentMethod(), accountId, snapshot);
        }

        public Task<ArenaData> GetArena(string accountId)
        {
            return CallMethod<ArenaData>(MethodBase.GetCurrentMethod(), accountId);
        }

        public Task<AcademyData> GetAcademy(string accountId)
        {
            return CallMethod<AcademyData>(MethodBase.GetCurrentMethod(), accountId);
        }
    }
}